### PR TITLE
GetPaymentDetailsLocal (with public note support)

### PR DESCRIPTION
**Pull request by patrickxb**

### CODE REVIEW

File: protocol/avdl/stellar1/remote.avdl [hunk 1]
```diff
@@ -127,7 +133,7 @@ protocol remote {
   array<PaymentSummary> recentPayments(keybase1.UserVersion caller, AccountID accountID, int limit);
 
   // txID can be either a keybase or stellar transaction ID.
-  PaymentSummary paymentDetail(keybase1.UserVersion caller, string txID);
+  PaymentDetails paymentDetails(keybase1.UserVersion caller, string txID);
```
[the remark this answers is not in the document]
Comment (patrickxb) [reply protocol/avdl/stellar1/remote.avdl]: Just trying to make it consistent across all the repos...

File: protocol/avdl/stellar1/local.avdl [hunk 2]
```diff
@@ -40,9 +40,11 @@ protocol local {
     ERROR_4,
     UNKNOWN_5
   }
+
+  @typedef("string") record PaymentID {}
   
   record PaymentLocal {
-    string id;                 // For relay payments, the funding tx.
+    PaymentID id;               // For relay payments, the funding tx.
```
[the remark this answers is not in the document]
Comment (patrickxb) [reply protocol/avdl/stellar1/local.avdl]: it used to be transaction id, but in a previous PR you asked for it to be opaque for frontend.  I thought PaymentID would hide for them what it was so they don't need to worry which ID we use behind the scenes.

File: go/stellar/relays/relays.go [hunk 3]
```diff
@@ -102,7 +102,7 @@ func Create(in Input) (res Output, err error) {
 		return res, err
 	}
 	sig, err := stellarnet.CreateAccountXLMTransaction(
-		senderSeed, relayAccountID, in.AmountXLM, in.SeqnoProvider)
+		senderSeed, relayAccountID, in.AmountXLM, "", in.SeqnoProvider)
```
[the remark this answers is not in the document]
Comment (patrickxb) [reply go/stellar/relays/relays.go]: done.

File: protocol/avdl/stellar1/local.avdl (hunk 2 above)
[the remark this answers is not in the document]
Comment (mlsteele) [reply protocol/avdl/stellar1/local.avdl]: You caught me. I was just projecting. Also I forgot this has to encompass non-keybase payments. If frontend hasn't said anything, my preference is not to add a third type. I guess just send the `stellar1.TransactionID`.

[the remark this answers is not in the document]
Comment (patrickxb) [reply protocol/avdl/stellar1/local.avdl]: reverted to transactionid

**Pull request merged**